python/paddle/v2/fluid/layers/nn.py

```diff
@@ -1971,3 +1971,56 @@ def sequence_reshape(input, new_dim):
         outputs={'Out': [out]},
         attrs={'new_dim': new_dim})
     return out
+
+
+def row_conv(input, future_context_size, param_attr=None, act=None):
+    """Row Conv Operator. This layer will apply lookahead convolution to
+    **input**. The input variable should be a 2D LoDTensor with shape [T, D].
+    Parameters with shape [future_context_size + 1, D] will be created. The math
+    equation of row convolution is as following:
+
+    .. math::
+        Out_{i} = \sum_{j = i} ^ {i + \\tau} X_{j} \odot W_{i - j}
+
+    In the above equation:
+
+    * :math:`Out_{i}`: The i-th row of output variable with shape [1, D].
+    * :math:`\\tau`: Future context size. Please note, the shape of
```

Comment:
Quick clarification: The value of $\tau$ is future_context_size + 1 right?

Reply:
<img width="811" alt="91345395-5355-49ee-aa49-ea3773afb587" src="https://user-images.githubusercontent.com/5782283/35208294-8d3c2a3c-ff82-11e7-888c-a2a84ecd0a7b.png">

Reply:
Sorry for this minor and maybe frustrating comment, but I think if $\tau = future_context_size$, then the formula in Line: [1983](https://github.com/PaddlePaddle/Paddle/pull/7575/files#diff-40735b79ee0c67af35d4e2611d3f76f1R1983) should have a $i + \tau + 1$. 

Maybe I misunderstood something?

Reply:
I think it is correct, we execute `future context size + 1` times of sum operations.

Reply:
Should be $i + \tau - i + 1$ == $\tau + 1$ ?

Reply:
Thanks.. I missed that